ui/page/components/components.go

```diff
@@ -117,113 +116,80 @@ func LayoutBalance(gtx layout.Context, l *load.Load, amount string) layout.Dimen
 // direction, balance, status.
 func LayoutTransactionRow(gtx layout.Context, l *load.Load, row TransactionRow) layout.Dimensions {
 	gtx.Constraints.Min.X = gtx.Constraints.Max.X
-	directionIconTopMargin := values.MarginPadding16
-
-	if row.Index == 0 && row.ShowBadge {
-		directionIconTopMargin = values.MarginPadding14
-	} else if row.Index == 0 {
-		// todo: remove top margin from container
-		directionIconTopMargin = values.MarginPadding0
-	}
 
 	wal := l.WL.MultiWallet.WalletWithID(row.Transaction.WalletID)
 
-	return layout.Inset{Top: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
-		return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
-			layout.Rigid(func(gtx C) D {
-				icon := l.Icons.ReceiveIcon
-				if row.Transaction.Direction == dcrlibwallet.TxDirectionSent {
-					icon = l.Icons.SendIcon
-				}
-				icon.Scale = 1.0
+	return decredmaterial.LinearLayout{
+		Orientation: layout.Horizontal,
+		Width:       decredmaterial.MatchParent,
+		Height:      gtx.Px(values.MarginPadding56),
+		Direction:   layout.W,
+		Padding:     layout.Inset{Left: values.MarginPadding16, Right: values.MarginPadding16},
+	}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			gtx.Constraints.Min.Y = gtx.Constraints.Max.Y
```

Comment:
why not add this as a field in LinearLayout?

Reply:
I noticed that `layout.Rigid` doesn't inherit the constraint property of the parent layout.